### PR TITLE
[SPARK-38196][SQL][FOLLOWUP] Make the testing CASE WHEN query more sane

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

Some testing CASE WHEN queries are not carefully written and do not make sense. In the future, the optimizer may get smarter and get rid of the CASE WHEN completely, and then we loose test coverage.

This PR updates some CASE WHEN queries to make them more sane.

### Why are the changes needed?

future-proof test coverage.

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

N/A

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala [hunk 1]
```diff
@@ -821,15 +821,13 @@ class JDBCV2Suite extends QueryTest with SharedSparkSession with ExplainSuiteHel
         |  COUNT(CASE WHEN SALARY > 11000 OR SALARY < 10000 THEN SALARY ELSE 0 END),
         |  COUNT(CASE WHEN SALARY >= 12000 OR SALARY < 9000 THEN SALARY ELSE 0 END),
         |  COUNT(CASE WHEN SALARY >= 12000 OR NOT(SALARY >= 9000) THEN SALARY ELSE 0 END),
-        |  MAX(CASE WHEN NOT(SALARY > 8000) AND SALARY >= 8000 THEN SALARY ELSE 0 END),
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala:824]: `NOT(SALARY > 8000) AND SALARY >= 8000` is the same as `SALARY = 8000`

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala [hunk 2]
```diff
@@ -821,15 +821,13 @@ class JDBCV2Suite extends QueryTest with SharedSparkSession with ExplainSuiteHel
...
-        |  MAX(CASE WHEN NOT(SALARY > 8000) OR SALARY > 8000 THEN SALARY ELSE 0 END),
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala:825]: `NOT(SALARY > 8000) OR SALARY > 8000` is always false

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala [hunk 3]
```diff
@@ -821,15 +821,13 @@ class JDBCV2Suite extends QueryTest with SharedSparkSession with ExplainSuiteHel
...
-        |  MAX(CASE WHEN NOT(SALARY > 8000) AND NOT(SALARY < 8000) THEN SALARY ELSE 0 END),
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala:826]: `NOT(SALARY > 8000) AND NOT(SALARY < 8000)` is the same as `SALARY = 8000`

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala [hunk 4]
```diff
@@ -821,15 +821,13 @@ class JDBCV2Suite extends QueryTest with SharedSparkSession with ExplainSuiteHel
...
+        |  MAX(CASE WHEN NOT(SALARY > 8000) AND SALARY >= 1000 THEN SALARY ELSE 0 END),
+        |  MAX(CASE WHEN NOT(SALARY > 8000) OR SALARY > 9000 THEN SALARY ELSE 0 END),
+        |  MAX(CASE WHEN NOT(SALARY > 8000) AND NOT(SALARY < 1000) THEN SALARY ELSE 0 END),
         |  MAX(CASE WHEN NOT(SALARY != 0) OR NOT(SALARY < 8000) THEN SALARY ELSE 0 END),
-        |  MAX(CASE WHEN NOT(SALARY > 8000 AND SALARY > 8000) THEN 0 ELSE SALARY END),
-        |  MIN(CASE WHEN NOT(SALARY > 8000 OR SALARY IS NULL) THEN SALARY ELSE 0 END),
-        |  SUM(CASE WHEN NOT(SALARY > 8000 AND SALARY IS NOT NULL) THEN SALARY ELSE 0 END),
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala:830]: `SALARY IS NOT NULL` can be inferred from `SALARY > 8000`

**Comment by beliefer** [comment]
@cloud-fan Thank you for the ping. I will take a look.

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala (hunk 4 above)
Comment (beliefer) [reply sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala:830]: It seems the infer only works in filter.

**Pull request closed without merging**